src/components/uploadpost/UploadPost.jsx

```diff
@@ -0,0 +1,204 @@
+  const [passwords, setPassword] = useState('');
+  const [content, setContent] = useState('');
+  const [time, setTime] = useState('');
+  const [participants, setParticipants] = useState('');
+  const [coverImage, setCoverImage] = useState('');
+  const [imageUrl, setImageUrl] = useState('');
+
+  const category = useSelector((state) => state.UploadPostSlice.category);
+  const lng = useSelector((state) => state.mapSlice.lng);
+  const lat = useSelector((state) => state.mapSlice.lat);
+
+  const navigate = useNavigate();
+
+  const handleClickConfirm = async () => {
+    let imageUploadUrl = null;
+
+    if (coverImage) {
+      imageUploadUrl = await uploadImage(coverImage);
+      console.log('Uploaded image URL:', imageUploadUrl);
+    }
+
+    const newObject = {
+      time: time,
+      title: title,
+      head_count: participants,
+      body: content,
+      user_id: nickname,
+      user_pw: passwords,
+      category,
+      map_lng: lng,
+      map_lat: lat,
+      image: imageUploadUrl // 이미지 URL을 설정합니다.
+    };
+
+    console.log(newObject);
+    await writePost(newObject);
+  };
+
+  const uploadImage = async (file) => {
+    const fileName = `${Date.now()}`;
+    const filePath = `${fileName}`;
+
+    try {
+      const { data, error } = await supabase.storage.from('images').upload(filePath, file);
+
+      if (error) {
+        throw error;
+      }
+
+      const { data: urldata, error: publicURLError } = await supabase.storage.from('images').getPublicUrl(filePath);
+      if (publicURLError) {
+        throw publicURLError;
+      }
+
+      return urldata.publicUrl;
+    } catch (error) {
+      console.error('Error uploading image:', error);
+      toast.error('이미지 업로드에 실패했습니다.');
+      return null;
+    }
+  };
+
+  const writePost = async (newObject) => {
+    try {
+      const { data, error } = await supabase.from('posts_test').insert(newObject);
+      if (error) {
+        throw error;
+      } else {
+        toast.success('글 작성이 완료되었습니다!', {
+          onClose: () => navigate('/'),
+          autoClose: 700,
+          hideProgressBar: true
+        });
+      }
+    } catch (e) {
+      console.error('Error writing post:', e);
+      toast.error('글 작성에 실패했습니다.');
+    }
+  };
+
+  const handleClickCancel = () => {
+    navigate(-1);
+  };
+
+  const handleImageChange = (e) => {
+    const file = e.target.files[0];
+    if (file) {
+      setCoverImage(file);
+      const imageUrl = URL.createObjectURL(file);
+      setImageUrl(imageUrl);
+      console.log(imageUrl);
+    }
+  };
+
+  return (
+    <StyleFormConatiner>
+      <div style={{ display: 'flex' }}>
+        <div style={{ display: 'flex', flexDirection: 'column', width: '100%' }}>
+          <StyledLabel htmlFor="title">제목</StyledLabel>
+          <StyleInput
+            type="text"
+            placeholder="제목 입력"
+            value={title}
+            onChange={(e) => setTitle(e.target.value)}
+            width="800px"
+            height="70px"
```

Comment:
```suggestion
            $width="800px"
            $height="70px"
```
styled component에만 내려줄 값이라면 이렇게 앞에 $를 작성해주어야 dom(진짜 input) 까지 전파가 되지 않습니다